src/main/java/seedu/address/logic/commands/schedule/FindCommonTimingCommand.java

```diff
@@ -55,51 +58,31 @@ public FindCommonTimingCommand(IsTagInPersonPredicate predicate, LocalDate date)
      * @param event event that needs to be accounted for
      */
     public void blockTimeSlots(int[] timeSlots, Event event) {
-        LocalTime startTime = event.getTime();
+        LocalTime startTime = event.getEventAtDate(event.getDate()).getTime();
         int hours = startTime.getHour();
         int minutes = startTime.getMinute();
         int startSlot = hours * 2;
         if (minutes >= 30) {
             startSlot += 1;
         }
 
-        LocalTime endTime = event.getEndTime();
+        LocalTime endTime = event.getEventAtDate(event.getDate()).getEndTime();
         int endHours = endTime.getHour();
         int endMinutes = endTime.getMinute();
         int endSlot = endHours * 2;
         if (endMinutes > 30) {
             endSlot += 1;
         }
 
-        for (int i = startSlot; i <= endSlot; i++) {
+        if (endTime == LocalTime.parse("00:00")) {
```

Comment:
```suggestion
        if (endTime == LocalTime.of(0,0) {
```
What you did is fine, but I think this would be better. I am assuming that parsing takes more time than using a static constructor.